File: pgm/ABC_Kotlin/ABC_383_A.kt
// import java.time.*

import kotlin.math.*

const val zero: Int = 0
const val space: String = " "

fun main() {
    val N: Int = readln().toInt()
    val T: MutableList<Int> = MutableList(N) { zero }
    val V: MutableList<Int> = MutableList(N) { zero }
    for (i: Int in zero until N) {
        val (t: Int, v: Int) = readln().split(space).map { it.toInt() }
        T[i] = t
        V[i] = v
    }
    var time: Int = zero
    var water: Int = zero
    for (i: Int in zero until N) {
        water = max(0, water - T[i] + time)
        water += V[i]
        time = T[i]
    }
    println(water)
    kotlin.system.exitProcess(0)
}

File: pgm/ABC_Dart/ABC_383_A.dart
// import 'dart:collection';
import 'dart:io';
import 'dart:math';

void main() {
  const int zero = 0;
  const String space = ' ';
  final int N = int.parse(stdin.readLineSync()!);
  final List<int> T = List.filled(N, zero);
  final List<int> V = List.filled(N, zero);
  for (int i = zero; i < N; i++) {
    final [int t, int v] = stdin
        .readLineSync()!
        .split(space)
        .map(int.parse)
        .toList();
    T[i] = t;
    V[i] = v;
  }
  int time = zero;
  int water = zero;
  for (int i = zero; i < N; i++) {
    water = max(zero, water - T[i] + time);
    water += V[i];
    time = T[i];
  }
  print(water);
  exit(0);
}
